addons/telemetry/telemetry.go

```diff
@@ -0,0 +1,158 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package telemetry
+
+import (
+	"k8s.io/utils/pointer"
+
+	"github.com/apache/camel-k/addons/telemetry/discovery"
+	v1 "github.com/apache/camel-k/pkg/apis/camel/v1"
+	traitv1 "github.com/apache/camel-k/pkg/apis/camel/v1/trait"
+	"github.com/apache/camel-k/pkg/trait"
+	"github.com/apache/camel-k/pkg/util"
+)
+
+// The Telemetry trait can be used to automatically publish tracing information to an OTLP compatible collector.
+//
+// The trait is able to automatically discover the telemetry OTLP endpoint available in the namespace (supports **Jaerger** in version 1.35+).
+//
+// The Telemetry trait is disabled by default.
+//
+// WARNING: The Telemetry trait can't be enabled at the same time as the Tracing trait.
+//
+// +camel-k:trait=telemetry.
+type Trait struct {
+	traitv1.Trait `property:",squash" json:",inline"`
+	// Enables automatic configuration of the trait, including automatic discovery of the telemetry endpoint.
+	Auto *bool `property:"auto" json:"auto,omitempty"`
+	// The name of the service that publishes telemetry data (defaults to the integration name)
+	ServiceName string `property:"service-name" json:"serviceName,omitempty"`
+	// The target endpoint of the Telemetry service (automatically discovered by default)
+	Endpoint string `property:"endpoint" json:"endpoint,omitempty"`
+	// The sampler of the telemetry used for tracing (default "on")
+	Sampler string `property:"sampler" json:"sampler,omitempty"`
+	// The sampler ratio of the telemetry used for tracing
+	SamplerRatio string `property:"sampler-ratio" json:"sampler-ratio,omitempty"`
+	// The sampler of the telemetry used for tracing is parent based (default "true")
+	SamplerParentBased *bool `property:"sampler-parent-based" json:"sampler-parent-based,omitempty"`
+}
+
+type telemetryTrait struct {
+	trait.BaseTrait
+	Trait `property:",squash"`
+}
+
+const (
+	propEnabled            = "propEnabled"
+	propEndpoint           = "propEndpoint"
+	propServiceName        = "propServiceName"
+	propSampler            = "propSampler"
+	propSamplerRatio       = "propSamplerRatio"
+	propSamplerParentBased = "propSamplerParentBased"
+)
+
+var (
+	telemetryProperties = map[v1.RuntimeProvider]map[string]string{
+		v1.RuntimeProviderQuarkus: {
+			propEndpoint:           "quarkus.opentelemetry.tracer.exporter.otlp.endpoint",
+			propServiceName:        "quarkus.opentelemetry.tracer.resource-attributes",
+			propSampler:            "quarkus.opentelemetry.tracer.sampler",
+			propSamplerRatio:       "quarkus.opentelemetry.tracer.sampler.ratio",
+			propSamplerParentBased: "quarkus.opentelemetry.tracer.sampler.parent-based",
+		},
+	}
+)
+
+// NewTelemetryTrait --.
```

Comment:
nitpick: let's write some meaningful comment for new code.

Ideally, we should rewrite every comment like this `// xxxxx --.` from the codebase.